src/cider/nrepl/middleware/pprint.clj

```diff
@@ -0,0 +1,30 @@
+(ns cider.nrepl.middleware.pprint
+  (:require [clojure.pprint :refer [pprint *print-right-margin*]]
+            [clojure.tools.nrepl.middleware :refer [set-descriptor!]]
+            [cider.nrepl.middleware.util.cljs :as cljs]))
+
+(def pprint-eval
+  (comp pprint eval))
+
+(defn wrap-pprint-reply
+  [h {:keys [right-margin session] :as msg}]
+  (when right-margin
+    (swap! session assoc #'*print-right-margin* right-margin))
+  (h (assoc msg :eval 'cider.nrepl.middleware.pprint/pprint-eval)))
+
+(defn wrap-pprint
+  [h]
```

Comment:
I'd suggest naming this `handler` for clarity's sake. We know what `h` stands for here, but others might be confused.